### PR TITLE
Migrate some components to hooks

### PR DESCRIPTION
**Pull request by joshwooding**
# Description of changes

Migrates Migrate Button, CloseButton, Select, PartnerLogoLink and Article Group to use hooks

# Issue Resolved

Works towards #845

### CODE REVIEW

**Comment by joshwooding** [comment]
cc @kylemh

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit c45d0ddc15af5619976113e5424002b6bd614f73

https://deploy-preview-858--operation-code-storybook.netlify.com

**Comment by joshwooding** [comment]
cc @kylemh

File: components/Button/__tests__/Button.test.js [hunk 1]
```diff
@@ -57,13 +57,13 @@ describe('Button', () => {
...
 
-  it('should send log to console when clickHandler is called in non-prod environment', () => {
+  it('should send log to console when button is clicked in non-prod environment', () => {
     /* eslint-disable no-console */
     console.log = jest.fn();
 
     const ButtonShallowInstance = shallow(<Button>Testing</Button>);
 
-    ButtonShallowInstance.instance().clickHandler();
+    ButtonShallowInstance.simulate('click');
```
Comment (joshwooding) [inline components/Button/__tests__/Button.test.js]: This now tests the functionality, not the implementation.

File: components/Press/PressLinks/ArticleGroup/__tests__/ArticleGroup.test.js [hunk 2]
```diff
@@ -27,22 +27,24 @@ describe('ArticleGroup', () => {
...
+          { title: 'Example3', url: 'https://example3.com' },
         ]}
         numberOfInitiallyVisibleLinks={1}
       />,
     );
 
-    ArticleGroupShallowInstance.instance().clickHandler();
+    expect(ArticleGroupShallowInstance.find('li')).toHaveLength(1);
+
+    ArticleGroupShallowInstance.find('button').simulate('click');
```
Comment (joshwooding) [inline components/Press/PressLinks/ArticleGroup/__tests__/ArticleGroup.test.js]: This now tests the functionality, not the implementation.

File: components/Button/__tests__/Button.test.js (hunk 1 above)
Comment (kylemh) [reply components/Button/__tests__/Button.test.js]: Yep yep! Thanks for this. Excited for #850 cuz it'll automatically resolve a bunch of these.

**Comment by kylemh** [comment]
Implemented a few things instead of wasting ur time with minor critique

**Pull request merged**